### PR TITLE
Use `rimraf` package instead of `rm -rf` in `preinstall` script

### PR DESCRIPTION
**Pull request by oltionchampari**
Closes phovea/generator-phovea#404

### Developer Checklist (Definition of Done)

* [x] Descriptive title for this pull request provided (will be used for release notes later)
* [ ] All acceptance criteria from the issue are met
* [x] Branch is up-to-date with the branch to be merged with, i.e., develop
* [x] Code is cleaned up and formatted
* [ ] Code documentation written
* [ ] Unit tests written
* [x] Build is successful
* [x] [Summary of changes](#summary-of-changes) written
* [ ] Wiki documentation written
* [x] Assign at least one reviewer
* [x] Assign at least one assignee
* [x] Add type label (e.g., *bug*, *enhancement*) to this pull request
* [x] Add next version label (e.g., `release: minor`) to this PR following [semver](https://semver.org/)

### Summary of changes
* Replaced
```diff
- "preinstall": "rm -rf node_modules && rm package-lock.json || true"
+ "preinstall": "rimraf node_modules package-lock.json || exit 0"
```
* When using `rimraf node_modules` in windows you get a warning since the package deletes itself. We could solve this if we are prepared to install `rimraf `globally.
* Replaced `|| true` (throws an error in windows) with `|| exit 0 `which works for both windows and linux.
* Tested the command both in windows and linux and the workspace installation works as expected.

### Notes
 Another windows incompatible script is:
`"build:python": "rm -rf build/source && find . -name '*.pyc' -delete && node buildPython.js && cp -r ./<%-name.toLowerCase()%> build/source/",`
Both the `rm -rf` and `cp` do not exist in windows. I do not know though if this needs to be windows compatible though.

### CODE REVIEW

**Review by thinkh — CHANGES_REQUESTED** [review]
@oltionchampari Thanks for the fix. I tried that new preinstall command with my PowerShell on Windows 10 (with 20H2) and got the following error (in German):

```
In Zeile:1 Zeichen:39
+ rimraf node_modules package-lock.json || exit 0
+                                       ~~
Das Token "||" ist in dieser Version kein gültiges Anweisungstrennzeichen.
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : InvalidEndOfLine
```

Researching the error, I've found that the pipeline operators `||` and `&&` have been introduced with PowerShell v7.0.0 (see https://github.com/PowerShell/PowerShell/pull/9849). Hence, they will only work with the most recent PowerShell version. 

My current PowerShell version running `$host.Version` is:

```
Major  Minor  Build  Revision
-----  -----  -----  --------
5      1      19041  610
```

It seems to be the standard unless you are updating the PowerShell manually.

@anita-steiner We need to decide if Windows users require the newest PowerShell or we change the preinstall command.

**Comment by anita-steiner** [comment]
> @oltionchampari Thanks for the fix. I tried that new preinstall command with my PowerShell on Windows 10 (with 20H2) and got the following error (in German):
> 
> ```
> In Zeile:1 Zeichen:39
> + rimraf node_modules package-lock.json || exit 0
> +                                       ~~
> Das Token "||" ist in dieser Version kein gültiges Anweisungstrennzeichen.
>     + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
>     + FullyQualifiedErrorId : InvalidEndOfLine
> ```
> 
> Researching the error, I've found that the pipeline operators `||` and `&&` have been introduced with PowerShell v7.0.0 (see [PowerShell/PowerShell#9849](https://github.com/PowerShell/PowerShell/pull/9849)). Hence, they will only work with the most recent PowerShell version.
> 
> @anita-steiner We need to decide if Windows users require the newest PowerShell or we change the preinstall command.

I will check the version at bayer

**Comment by thinkh** [comment]
Btw there are a lot of still other scripts that use the pipeline operators (`&&` and `||`). We might need to change it as well (depending on the PowerShell version).

**Comment by anita-steiner** [comment]
so we need a solution for ps version >= 5.1

**Review by thinkh — APPROVED** [review]
I tested this PR again using PowerShell 5.1 to setup a clean TACO workspace with `yo phovea:setup-workspace Caleydo/taco_product -b develop` and it seems to work.

After the workspace is complete I ran `npm install` again and got the following (shortened) output:

```
PS D:\test\taco> npm i

> phovea_workspace@0.0.1 preinstall D:\test\taco
> rimraf node_modules package-lock.json || exit 0

Der Befehl "rimraf" ist entweder falsch geschrieben oder
konnte nicht gefunden werden.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated fsevents@2.1.3: Please update to v 2.2.x
npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142

[...]

> ejs@2.7.4 postinstall D:\test\taco\node_modules\ejs
> node ./postinstall.js

Thank you for installing EJS: built with the Jake JavaScript build tool (https://jakejs.com/)


> nodemon@1.19.4 postinstall D:\test\taco\node_modules\nodemon
> node bin/postinstall || exit 0


> node-sass@4.14.1 postinstall D:\test\taco\node_modules\node-sass
> node scripts/build.js

Binary found at D:\test\taco\node_modules\node-sass\vendor\win32-x64-72\binding.node
Testing binary
Binary is fine

> phovea_workspace@0.0.1 postinstall D:\test\taco
> npm-run-all --parallel delete-dependencies:*


> phovea_workspace@0.0.1 delete-dependencies:taco D:\test\taco
> cd taco && npm run delete-dependencies


> taco@5.0.1-SNAPSHOT delete-dependencies D:\test\taco\taco
> rimraf node_modules

npm notice created a lockfile as package-lock.json. You should commit this file.

[...]

added 1455 packages from 981 contributors and audited 1459 packages in 45.746s
found 1 low severity vulnerability
  run `npm audit fix` to fix them, or `npm audit` for details
PS D:\test\taco>
```

Furthermore, I researched again if other people have the same problem and found [this nodemon issue](https://github.com/remy/nodemon/issues/1663) where some people could reproduce it and some not. At the end the issue was closed as it could not be reproduced.

Long story short: This PR seems to work, but we have other places where we are using Linux specific commands that do not work with PowerShell (see also #429).

**Review by anita-steiner — APPROVED** [review]
